internal/host/host.go

```diff
@@ -1229,13 +1229,13 @@ func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.D
 	return false, nil
 }
 
-// This function recommends a role for a given host based on these criteria:
-//  1. if there are not enough masters and the host has enough capabilities to be
+// selectRole recommends a role for a given host based on these criteria:
+//   - if there are not enough masters and the host has enough capabilities to be
 //     a master the function select it to be a master
-//  2. if there are enough masters, or it is a day2 host, or it has not enough capabilities
-//     to be a master the function select it to be a  worker
-//  3. in case of missing inventory or an internal error the function returns auto-assign
-func (m *Manager) selectRole(ctx context.Context, h *models.Host, db *gorm.DB) (models.HostRole, error) {
+//   - if there are enough masters, or it is a day2 host, or it does not not have enough capabilities
```

Comment:
Just FYI, in the future, it is possible some day2 hosts will be masters.

Reply:
https://github.com/openshift/assisted-service/pull/6917#discussion_r1839668842